### ISSUE TITLE
Which parts of the training process are frozen?

### ISSUE THREAD

**Issue by bat1115** [sjtuplayer/anomalydiffusion#51]
![image](https://github.com/sjtuplayer/anomalydiffusion/assets/58722076/ab226485-2f0e-40c7-bf55-277db17604a1)
![image](https://github.com/sjtuplayer/anomalydiffusion/assets/58722076/da65aab8-81d9-4dba-b336-bc98f82be407)
![image](https://github.com/sjtuplayer/anomalydiffusion/assets/58722076/9135e697-c950-404d-8257-79a55936df9f)
Are the Denoised Unet of LDM and BERTEmbedder all trained? What is the use of configure_optimizers?

**Comment by sjtuplayer** [comment]
No, both the Denoised Unet of LDM and BERTEmbedder are not trained. The codebase is borrowed from textual inversion, so there are some other options in configure_optimizers. But only the first if branch is valid in our code.

**Comment by bat1115** [comment]
> No, both the Denoised Unet of LDM and BERTEmbedder are not trained. The codebase is borrowed from textual inversion, so there are some other options in configure_optimizers. But only the first if branch is valid in our code.

What about configure_opt_embedding( ) and configure_opt_model( )? Are they run in the training process?

**Comment by bat1115** [comment]
> > No, both the Denoised Unet of LDM and BERTEmbedder are not trained. The codebase is borrowed from textual inversion, so there are some other options in configure_optimizers. But only the first if branch is valid in our code.
> 
> What about configure_opt_embedding( ) and configure_opt_model( )? Are they run in the training process?

My question is how does BERTEmbedder guarantee the effect of z,position = self.transformer(tokens, return_embeddings=True, **kwargs) if the parameters are not updated? Because self.transformer = TransformerWrapper() is not a pre-trained model.
